### PR TITLE
Improve shift-tab experience in the toolbar

### PR DESCRIPTION
**Pull request by iainx**
Correctly implement shift-tab behaviour in the selector
Correctly shift tab into the toolbar, landing on the searchbar instead of the run button
Shift tab out of the toolbar, back into the Gtk widgets.

Fix VSTS #648190

### CODE REVIEW

File: main/src/addins/WindowsPlatform/WindowsPlatform/MainToolbar/WPFToolbar.cs [hunk 1]
```diff
@@ -306,6 +306,11 @@ public void Focus (System.Action exitAction)
 			exitAction ();
 		}
 
+		public void Focus (Gtk.DirectionType direction, Action<Gtk.DirectionType> exitAction)
+		{
+			FocusSearchBar ();
+			exitAction (Gtk.DirectionType.TabForward);
```
Comment (Therzok) [inline main/src/addins/WindowsPlatform/WindowsPlatform/MainToolbar/WPFToolbar.cs]: `exitAction (direction)`?

File: main/src/addins/MacPlatform/MainToolbar/SelectorView.cs [hunk 2]
```diff
@@ -615,6 +628,17 @@ void SetSelection ()
 
 			public override bool BecomeFirstResponder ()
 			{
+				var currentEvent = NSApplication.SharedApplication.CurrentEvent;
+				// Check if the currentEvent that caused us to become first responder is a Tab or a Reverse Tab
+				if (currentEvent.Type == NSEventType.KeyDown) {
+					if (currentEvent.KeyCode == 0x30) {
```
Comment (Therzok) [inline main/src/addins/MacPlatform/MainToolbar/SelectorView.cs]: What is `0x30`

Comment (iainx) [reply main/src/addins/MacPlatform/MainToolbar/SelectorView.cs]: Tab, I'll add a comment

Comment (slluis) [reply main/src/addins/MacPlatform/MainToolbar/SelectorView.cs]: Better add a constant

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components.MainToolbar/IMainToolbarView.cs [hunk 3]
```diff
@@ -262,8 +262,11 @@ public interface IButtonBarView
 	/// </summary>
 	public interface IMainToolbarView : IRunButtonView, ISelectorView, ISearchEntryView, IStatusBarView, IButtonBarView
 	{
+		[Obsolete]
 		void Focus ();
+		[Obsolete]
 		void Focus (Action exitAction);
+		void Focus (Gtk.DirectionType direction, Action<Gtk.DirectionType> exitAction);
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components.MainToolbar/IMainToolbarView.cs]: This interface is not intended to be implemented or directly used by extensions, so feel free to just break the API, no need to obsolete the methods.

**Pull request merged**